staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go

```diff
@@ -70,7 +70,7 @@ var ValidateServiceAccountName = NameIsDNSSubdomain
 // value if is a dash.
 func maskTrailingDash(name string) string {
 	if strings.HasSuffix(name, "-") {
-		return name[:len(name)-2] + "a"
```

Comment:
"-2" is wrong. On the other hand, The validation should fail "-" as a prefix name, but now "-" will be rewritten as "a" and will pass the validation. Could you handle the case name="-"? Also, could you add tests to prevent regression?